dhctl/cmd/dhctl/commands/converge.go

```diff
@@ -29,6 +31,31 @@ import (
 	"github.com/deckhouse/deckhouse/dhctl/pkg/terraform"
 )
 
+func getCacheIdentityFromKubeconfig(
+	kubeconfigPath string,
+	kubeconfigContext string,
+) (string, error) {
+	builder := strings.Builder{}
+	builder.WriteString("kubeconfig")
+
+	kubeconfigName := filepath.Base(kubeconfigPath)
+	if kubeconfigName == "." || kubeconfigName == ".." || kubeconfigName == "/" {
+		return "", fmt.Errorf("incorrect kubeconfig path")
+	}
+
+	builder.WriteString("-")
+	builder.WriteString(kubeconfigName)
+
+	if kubeconfigContext == "" {
+		return builder.String(), nil
+	}
+
+	builder.WriteString("-")
+	builder.WriteString(kubeconfigContext)
+
+	return builder.String(), nil
```

Comment:
AFAIR, this should be sha256 encoded, but maybe this is encoded later.

Reply:
Yep it will be encoded in another place